security/tpm/src/tpm.ts

```diff
@@ -3,22 +3,428 @@
 
 'use strict';
 import { errors } from 'azure-iot-common';
+import * as machina from 'machina';
+import * as tss from 'tss.js';
+import { Tpm, TPM_HANDLE, TPM_ALG_ID, TPM_RC, TPM_PT, TPMA_OBJECT, TPMT_PUBLIC, TPM2B_PRIVATE } from 'tss.js';
+import * as crypto from 'crypto';
+import base32Encode = require('base32-encode');
+
+import * as dbg from 'debug';
+
+const debug = dbg('azure-iot-security-tpm:tpmclient');
+
+
+const aes128SymDef = new tss.TPMT_SYM_DEF_OBJECT(TPM_ALG_ID.AES, 128, TPM_ALG_ID.CFB);
+
+const ekPersHandle: TPM_HANDLE = new TPM_HANDLE(0x81010001);
+const srkPersHandle: TPM_HANDLE = new TPM_HANDLE(0x81000001);
+const idKeyPersHandle: TPM_HANDLE = new TPM_HANDLE(0x81000100);
+
+// Template of the Endorsement Key
+const ekTemplate = new TPMT_PUBLIC(TPM_ALG_ID.SHA256,
+  TPMA_OBJECT.restricted | TPMA_OBJECT.decrypt | TPMA_OBJECT.fixedTPM | TPMA_OBJECT.fixedParent | TPMA_OBJECT.adminWithPolicy | TPMA_OBJECT.sensitiveDataOrigin,
+  new Buffer('837197674484b3f81a90cc8d46a5d724fd52d76e06520b64f2a1da1b331469aa', 'hex'),
```

Comment:
>837197674484b3f81a90cc8d46a5d724fd52d76e06520b64f2a1da1b331469aa [](start = 14, length = 64)

?

Reply:
See above.

---
In reply to: [157881650](https://github.com/Azure/azure-iot-sdk-node/pull/185#discussion_r157881650) [](ancestors = 157881650)